### PR TITLE
Blog App : Forms

### PR DESCRIPTION
**Pull request by torobucci**
-  Create a method called `current_user` in your ApplicationController to make current user data available to all controllers.
    -  It will return the first user from the database.
- Create forms to perform the following functions:
    - Creates a Post on behalf of the `current_user` 
    - Create a comment on behalf of the `current_user`
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -9,4 +9,21 @@ def show
...
+    @post.author_id = current_user.id
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+    if @post.save
+      redirect_to user_posts_path(current_user, @post.id), notice: 'Post created successfully.'
+    else
+      flash.now[:error] = 'Error: Post could not be saved'
+      render :new
+    end
+  end
```
Comment (codecaiine) [inline app/controllers/posts_controller.rb:28]: - [x] [OPTIONAL] As a suggestion I would like to recommend you add `the params method` , you can read more in this [lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides )  👍 
```ruby
   private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CommentsController < ApplicationController
+  def create
```
Comment (codecaiine) [inline app/controllers/comments_controller.rb:7]: - [ ] According to the project requirement,  the comments controller should have` new `and `create` actions similar to the `posts_controllers.rb`. The **new** action is missing here. Please fix it to make your app works as expected 👍

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.new(params.require(:comment).permit(:text))
+    @comment.author = current_user
+    if @comment.save
+      redirect_to user_post_url(@post.author, @post), notice: 'Comment created successfully.'
+    else
+      flash.now[:error] = 'Error: Comment could not be saved'
+      redirect_to user_post_url(@post.author, @post)
+    end
+  end
```
Comment (codecaiine) [inline app/controllers/comments_controller.rb:17]: - [ ] [OPTIONAL] As a suggestion I would like to recommend you add `the params method` , you can read more in this [lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)  👍 

```ruby
  private 
  def comment_params
    params.require(:comment).permit(:text)
  end
```

File: app/views/posts/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<div>
+ <h3>Create New Post</h3>
+  <%= form_with(model:  @post, url: user_posts_path, local: true) do |form| %>
+    <%= form.label :title %>
+    <%= form.text_field :title, name: 'post[title]' %>
+    <%= form.label :text %>
+    <%= form.text_area :text, name: 'post[text]' %>
+    <%= form.submit 'Create Post' %>
+  <% end %>
+</div>
```
Comment (codecaiine) [inline app/views/posts/new.html.erb:10]: - [ ] Good job so  far 👏  But you should also create a form in the **views/comments/new.html.erb** file which with input for creating comments also please create all appropriate routes in order to make your project as expected 👍

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @torobucci , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Good actions implementation ✔️ 
- Good form setup ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- You meet all the requested changes. :heavy_check_mark: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**